### DISCUSSION TITLE
How to solve MAMP warning about wrong encoding, conversion from UTF-8-MAC to UTF-8?

### DISCUSSION THREAD

**Discussion by pawell0** [datenstrom/community#937] [Ask a question] [answered]
Hi,

When I swiched the PHP version to 8.2.0 on my MAMP i have the warning:
Warning: iconv(): Wrong encoding, conversion from "UTF-8-MAC" to "UTF-8" is not allowed in /Users/(username)/Sites/yellow/system/extensions/core.php on line 1538

M1 Mac, Ventura 13.6.4, MAMP latest universal, Yellow latest and updated.

What should I do or what do I incorrectly? Could this also be the reason that one of the pages with a title containing special characters has disappeared from the menu? More: I swtched on the Yellow debug mode and this message is present everytime I'm trying to open a page with a title containing special characters, no matter which version of PHP is used. All the pages with titles containing special characters are now missing with error: "File not found". Example report:

`YellowCore::sendPage Cache-Control: no-cache, no-store
YellowCore::sendPage Content-Type: text/html; charset=utf-8
YellowCore::sendPage Content-Modified: Thu, 01 Jan 1970 00:00:00 GMT
YellowCore::sendPage Last-Modified: Mon, 05 Feb 2024 05:43:48 GMT
YellowCore::sendPage language:pl layout:error theme:paris parser:markdown
YellowCore::processRequestError file:content/2-blog/z-prądem-czy-pod-prąd.md
YellowCore::request status:404 time:38 ms`

My guess is that the MAMP is douing something wrong. I never had this kind of errors on my previous MAMP (6.8) installation on my previous Intel Mac.

I'm using this Yellow instance to edit and mainain my website locally, later on the changes are implemented to the Yellow instance online on the other commercial server ...

**Comment by pawell0** [comment, marked as answer]
OK! I found the solution. I will modify all my URL adresses according to:
https://github.com/datenstrom/yellow/discussions/326

I will remove special characters in all page adresses (names of the *.MD files) and I will add the `TitleSlug` in the files ... I wish I had thought of this earlier. 😆 Thanks for implementig that feature.
BTW - there is a simple remedy to further such problems. In the `/yellow/contennt/shared` directory I found templates for new site files. These are: `page-new-blog.md` and `page-new-default.md`. Just below the `Title` add `TitleSlug` in them so that its filling is "suggested" when creating a new page, instead of adding it manually after the online editor appears (In case you forgot ...).

**Comment by markseuffert** [comment]
Please note the warning comes from the function `lookup->normaliseUnicode()`. The core has to convert file names on MacOS from UTF-8 NFD into UTF-8 NFC. This is what's commonly used in software such as web browsers and text editors. Should the normalisation of Unicode characters not be possible, then you can no longer use European special characters in file names. You can experiment with different PHP versions and workarounds, have a look at datenstrom/community#864, datenstrom/community#774, datenstrom/community#372.

**Reply by pawell0** [reply]
Thanks, Mark. My "remedy" is working well ... There is no need to look further. It was my mistake for not thinking of this earlier.

**Reply by markseuffert** [reply]
Thanks, Pawel. Good to hear you found a solution that works for you. If you have some time can you test different PHP versions on your local machine, I am curious if this gets rid of the warning.

**Reply by pawell0** [reply]
Mark, I tested the both of the PHP versions available (8.2.0 and 7.4.33) in my MAMP (free version). After adding `TitleSlug` on the pages and removing special characters in the *.md file names of the pages, everything works OK.
